### ISSUE TITLE
Improve mime type sniffing logic

### ISSUE THREAD

**Issue by kislyuk** [HumanCellAtlas/dcp-cli#50]
```
hca upload --replica aws --staging-bucket $DSS_S3_BUCKET_TEST --file-or-dir data-bundle-examples/10X_v2/pbmc8k > upload.json
Traceback (most recent call last):
  File "/Users/akislyuk/projects/data-store/v3/bin/hca", line 11, in <module>
    hca.main()
  File "/Users/akislyuk/projects/data-store/v3/lib/python3.6/site-packages/hca/__init__.py", line 25, in main
    response = cli.make_request(sys.argv[1:])
  File "/Users/akislyuk/projects/data-store/v3/lib/python3.6/site-packages/hca/cli.py", line 63, in make_request
    return command.run_from_cli(args)
  File "/Users/akislyuk/projects/data-store/v3/lib/python3.6/site-packages/hca/api/composite_commands/upload.py", line 192, in run_from_cli
    return cls.run(args)
  File "/Users/akislyuk/projects/data-store/v3/lib/python3.6/site-packages/hca/api/composite_commands/upload.py", line 208, in run
    filename_key_list = cls._upload_files(args, staging_bucket)
  File "/Users/akislyuk/projects/data-store/v3/lib/python3.6/site-packages/hca/api/composite_commands/upload.py", line 94, in _upload_files
    file_uuids, uploaded_keys = upload_to_cloud(files_to_upload, staging_bucket, args['replica'], from_cloud)
  File "/Users/akislyuk/projects/data-store/v3/lib/python3.6/site-packages/hca/upload_to_cloud.py", line 86, in upload_to_cloud
    "hca-dss-content-type": _mime_type(fh.raw.name)
  File "/Users/akislyuk/projects/data-store/v3/lib/python3.6/site-packages/hca/upload_to_cloud.py", line 29, in _mime_type
    raise RuntimeError("Can't discern mime type")
RuntimeError: Can't discern mime type
smoketest.py: Exit status 1 while running "hca upload --replica aws --staging-bucket $DSS_S3_BUCKET_TEST --file-or-dir data-bundle-examples/10X_v2/pbmc8k > upload.json". Stopping.
```

**Comment by kislyuk** [comment]
The crash is fixed in #51. Repurposing this issue to track improving the mime type sniffing logic to use something like libmagic, and assign HCA-specific types (hypothetical examples: `application/json; hca-metadata/assay`, `application/octet-stream; sequence-alignment/bam`).

**Comment by kislyuk** [comment]
@sampierson fixed this in #76.
